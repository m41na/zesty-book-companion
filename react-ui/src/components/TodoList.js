import React from 'react';
import './css/todos.css';

class TodoList extends React.Component{
    constructor(props){
        super(props);
        this.state = {
            task: ''
        };
    }
    componentDidMount(){
        this.props.retrieveTasks();
    }
    onCreateTask(e){
        e.preventDefault();
        this.props.createTask(this.state.task);
        this.setState({task: ''});
    }
    onUpdateDone(name, done){
        this.props.updateDone(name, !done);
    }
    onDeleteTask(name){
        this.props.deleteTask(name);
    }
    onChange(e){
        let value = e.target.value;
        this.setState({[e.target.name]: value});
    }
    render(){
        const {todos} = this.props;

        return (
            <div className="wrapper">
                <div id="myDIV" className="header">
                    <h2>My To Do List</h2>
                    <form id="tasks_form" onSubmit={this.onCreateTask.bind(this)}>
                        <input type="text" name="task" placeholder="Title..." value={this.state.task} onChange={this.onChange.bind(this)} />
                        <button type="submit" className="addBtn">Add</button>
                    </form>
                </div>

                <ul id="myUL">
                    {todos && todos.map(e=>
                    <li key={e.name} data-name={e.name} className={e.completed?'checked':''} >
                        <span onClick={()=>this.onUpdateDone(e.name, e.completed)}>{e.name}</span> 
                        <span className="close" onClick={()=>this.onDeleteTask(e.name)}>x</span>
                    </li>
                    )}
                </ul> 
            </div>
        );
    }
}
export default TodoList;