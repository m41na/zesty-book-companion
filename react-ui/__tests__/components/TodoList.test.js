import React from 'react';
import {shallow} from 'enzyme';
import TodoList from '../../src/components/TodoList';

const initialProps = {
    todos: [{'name': 'biking', completed: false},{name: 'hiking', completed: true}],
    retrieveTasks(){
        console.log('retrieveTasks');
    },
    createTask(){
        console.log('createTask');
    },
    updateDone(){
        console.log('updateDone');
    },
    deleteTask(){
        console.log('deleteTask');
    }
};

describe('TodoList behavior', () => {

    describe('render()', () => {
        
        it('should render the component', () => {
            const component = shallow(<TodoList {...initialProps}/>);
            expect(component).toMatchSnapshot();        

            //assert list has two tasks
            const list = component.find("#myUL li");
            expect(list).toHaveLength(2);

            //assert first list item is 'biking'
            let text = component.first().find("span").first().text();
            expect(text).toBe('biking');
        });
    });
});