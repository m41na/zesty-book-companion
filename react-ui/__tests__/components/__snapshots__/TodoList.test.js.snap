// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TodoList behavior render() should render the component 1`] = `
<div
  className="wrapper"
>
  <div
    className="header"
    id="myDIV"
  >
    <h2>
      My To Do List
    </h2>
    <form
      id="tasks_form"
      onSubmit={[Function]}
    >
      <input
        name="task"
        onChange={[Function]}
        placeholder="Title..."
        type="text"
        value=""
      />
      <button
        className="addBtn"
        type="submit"
      >
        Add
      </button>
    </form>
  </div>
  <ul
    id="myUL"
  >
    <li
      className=""
      data-name="biking"
      key="biking"
    >
      <span
        onClick={[Function]}
      >
        biking
      </span>
      <span
        className="close"
        onClick={[Function]}
      >
        x
      </span>
    </li>
    <li
      className="checked"
      data-name="hiking"
      key="hiking"
    >
      <span
        onClick={[Function]}
      >
        hiking
      </span>
      <span
        className="close"
        onClick={[Function]}
      >
        x
      </span>
    </li>
  </ul>
</div>
`;
